app/handlers/sessions_help.rb

```diff
@@ -16,15 +16,52 @@ def authorized?
   end
 
   def handle
-    username_or_email = help_params.username_or_email.strip
-    user = User.find_by_username(username_or_email) ||
-           ContactInfo.find_by_value(username_or_email).try(:user)
-    if user.nil?
-      fatal_error(code: :user_not_found,
-                  message: 'We did not find an account with the username or email you provided.',
-                  offending_inputs: [:username])
+    multiple_users = matching_users.count > 1
+
+    matching_users.each do |matching_user|
+      user = matching_user[:user]
+
+      code = run(GeneratePasswordResetCode, user.identity).outputs[:code] \
+        unless user.identity.nil?
+
+      email_addresses = matching_user[:email_addresses]
+
+      email_addresses.each do |email_address|
+        SignInHelpMailer.sign_in_help(
+          user: user,
+          email_address: email_address,
+          reset_password_code: code,
+          multiple_emails_per_user: email_addresses.count > 1,
+          multiple_users: matching_users.count > 1
+        ).deliver
+      end
     end
-    email_addresses = user.contact_infos.email_addresses
+  end
+
+  def matching_users
+    if @matching_users.nil?
+      username_or_email = help_params.username_or_email.strip
+
+      @matching_users = matching_users_by_username(username_or_email) ||
+                        matching_users_by_email(username_or_email)
+
+      if @matching_users.blank?
+        fatal_error(code: :user_not_found,
```

Comment:
I think this code will aways be true and then raise a `fatal_error`. Since the if statement block is checking `.nil?`, then `.blank?` will also be true since it will match both `.nil?` and `.empty?`